carmajava/build.gradle

```diff
@@ -90,12 +115,45 @@ subprojects {
     
 }
 
+// Task to collect all generated jacoco reports and place them in the top level build folder
+// Based off method described here https://gist.github.com/aalmiray/e6f54aa4b3803be0bcac
+task jacocoCombinedReport(type: org.gradle.testing.jacoco.tasks.JacocoReport) {
+    // Requires submodule have test task
+    dependsOn = subprojects.test
+    // Set sources and classes to defaults
+    additionalSourceDirs = files(subprojects.sourceSets.main.allSource.srcDirs)
+    sourceDirectories = files(subprojects.sourceSets.main.allSource.srcDirs)
+    classDirectories =  files(subprojects.sourceSets.main.output)
+    executionData = files(subprojects.jacocoTestReport.executionData)
+    // Turn on XML reports.
+    // The XML report is required by SonarCloud
+    // The default location will put the reports  CARMAPlatform/carmajava/build/reports/jacoco/jacocoCombinedReport
+    reports {
+        html.enabled = false
+        xml.enabled = true
+        csv.enabled = false
+    }
+    // The onlyIf and doFirst lines here cause this to only run for submodules with tests
+    onlyIf = {
+        true
+    }
+    doFirst {
+        executionData = files(executionData.findAll {
+            it.exists()
+        })
+    }
+}
+
 task testReport(type: TestReport) {
     destinationDir = file("$buildDir/reports/allTests")
     // Include the results from the `test` task in all subprojects
     reportOn subprojects*.test
 }
 
+task testCoverageReport() {
```

Comment:
Is this supposed to be empty? Unless it matters that it's defined, can we just delete it?